dbldatagen/utils.py

```diff
@@ -196,9 +196,9 @@ def parse_time_interval(spec):
             hours = time_value
         elif time_type in ["minutes", "minute"]:
             minutes = time_value
-        elif time_type in ["seconds", "seconds"]:
+        elif time_type in ["seconds", "second"]:
             seconds = time_value
-        elif time_type in ["microseconds", "microseconds"]:
+        elif time_type in ["microseconds", "microsecond"]:
             microseconds = time_value
         elif time_type in ["milliseconds", "millisecond"]:
             milliseconds = time_value
```

Comment:
huge chain of if/elif is usually a sign that you have an improperly typed input. These operations share same logic, but are repetative. 
You'll better write something like:
```
unit_measure = ["second", "millisecond", "microsecond"]
for unit in unit_measure:
    if some_input in unit, f"{unit}s":
        do_something_with_it(some_input)
```

Reply:
logic is not the same - different component of time being assigned in each case  - but agree on general principal